pyvista/core/composite.py

```diff
@@ -460,23 +463,40 @@ def __setitem__(self, index: Union[Tuple[int, Optional[str]], int, str], data: D
             i, name = cast(int, index), None
         if data is not None and not is_pyvista_dataset(data):
             data = wrap(data)
+
         if i == -1:
             self.append(data)
             i = self.n_blocks - 1
```

Comment:
Unrelated to this PR, but it bugs me a bit that the execution path is
1. `multi[-1] = mesh`
2. `multi.append(mesh)`
3. `multi[n_blocks] = mesh`
with execution then continuing in the first entry in `__setitem__`.

Do you think we could just handle the `-1` case directly in `__setitem__`? Does the detour through `append` give us anything here?